spec/controllers/audio_files_controller_spec.rb

```diff
@@ -0,0 +1,27 @@
+require "rails_helper"
+
+RSpec.describe AudioFilesController do
+  describe "POST #create" do
+    it "creates a new activity" do
+      allow(Activity).to receive(:create!)
+
+      user = FactoryGirl.create(:user)
+      mock_audio_file = double(AudioFile, save: true)
+      mock_audio_file_relation = double(:audio_files, new: mock_audio_file)
+      allow(user).to receive(:audio_files).and_return(mock_audio_file_relation)
```

Comment:
There's a bit happening here that is part of the set up of the test that we don't necessarily care about in our execution of the test. Consider extracting a helper method such as:

```ruby

def stub_relation(model_instance, relation_name)
  mock_instance = double(:instance, save: true)
  mock_relation = double(:relation, new: mock_instance)
  allow(model_instance).to receive(:relation_name).and_return(mock_relation)

   mock_instance
end
```

Then this can be used as:

```ruby
it "creates a new activity" do
  allow(Activity).to receive(:create)
  user = FactoryGirl.create(:user)
  mock_audio_file = stub_relation(user, :audio_files)

  sign_in_as(user)
  # ....
```